datalad/utils.py

```diff
@@ -397,3 +397,22 @@ def assure_dir(*args):
         os.makedirs(dirname)
     return dirname
 
+def chpwd(path):
+    """Wrapper around os.chdir which also adjusts environ['PWD']
+
+    The reason is that otherwise PWD is simply inherited from the shell
+    and we have no ability to assess directory path without dereferencing
+    symlinks
+    """
+    os.chdir(path)  # for grep people -- ok, to chdir here!
+    os.environ['PWD'] = path
```

Comment:
This seems inapproriate. From http://pubs.opengroup.org/onlinepubs/009695399/basedefs/xbd_chap08.html

    This variable shall represent an absolute pathname of the current working directory. It shall not contain any
    filename components of dot or dot-dot

As far as I can see, nothing prevent `path` from being relative in your implementation.